### PR TITLE
Implementation of Plugin Tasks

### PR DESCRIPTION
**Pull request by gusingh-r7**
### Description
- Add implementation and endpoints to support tasks.

### Types of changes

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to change)

### Checklist:

- [ ] I have updated the documentation accordingly (if changes are required).
- [ ] I have tested the changes locally.

### CODE REVIEW

**Review by zyoutz-r7 — APPROVED** [review]
Looking good! Marked as DO NOT MERGE until we are ready to get it in.

**Review by lmilby-r7 — DISMISSED** [review]

**Review by zyoutz-r7 — DISMISSED** [review]
Want to make sure the tests all pass before merging but these changes look great 👍

File: insightconnect_plugin_runtime/variables.py [hunk 1]
```diff
@@ -83,3 +83,24 @@ def validate(self, parameters):
...
+        if should_validate:
+            self.validate(self.parameters)
+
+    def validate(self, parameters):
+        """ Validate variables """
+        validate(parameters, self.schema)
+
+    def sample(self):
+        """ Sample object """
+        return util.sample(self.schema)
```
Comment (mrinehart-r7) [inline insightconnect_plugin_runtime/variables.py:106]: Input, Output, and State all have the same `__init__`, `set`, `validate`, and `sample` functions - should/could that all be moved to a parent class that those three classes inherit from and override as necessary?

**Review by mrinehart-r7 — CHANGES_REQUESTED** [review]

File: tests/unit/test_helpers.py [hunk 2]
```diff
@@ -251,17 +251,6 @@ def test_encode_file_success():
...
-
-
-def test_check_url_modified_false():
-    assert not helper.check_url_modified("https://httpstat.us/304")
-
-
-def test_check_url_modified_true():
-    assert helper.check_url_modified("https://httpstat.us/200")
-
-
```
Comment (gusingh-r7) [inline tests/unit/test_helpers.py:264]: Removing these test cases as https://httpstatus does not seem working.

**Review by zyoutz-r7 — APPROVED** [review]
🔥

**Review by mrinehart-r7 — APPROVED** [review]

**Review by mrinehart-r7 — APPROVED** [review]

**Pull request merged**